test/cql-pytest/test_guardrail_replication_strategy.py

```diff
@@ -0,0 +1,92 @@
+import pytest
+from util import unique_name
+from cassandra.protocol import ConfigurationException
+
+def create_ks_and_assert_warnings_count(cql, ks_opts, warnings_count = 0, fails_count = 0):
+    keyspace = unique_name()
+    if fails_count:
+        with pytest.raises(ConfigurationException):
+            cql.execute_async("CREATE KEYSPACE " + keyspace + " " + ks_opts).result()
+        return
+    else:
+        response_future = cql.execute_async("CREATE KEYSPACE " + keyspace + " " + ks_opts)
+        response_future.result()
+        if warnings_count:
+            assert len(response_future.warnings) == warnings_count
+        else:
+            assert response_future.warnings is None
+        cql.execute("DROP KEYSPACE " + keyspace)
+
+def test_replication_strategy_warn_list(cql, this_dc):
+    """
+    Test that placing given replication strategy on a guardrail's replication_strategy_warn_list
+    actually generates a warning
+    """
+
+    # First let's clear the fail & warn lists to see if we don't generate a warning by accident when creating a ks
+    cql.execute(f"UPDATE system.config SET value = '' where name = 'replication_strategy_warn_list'")
+    cql.execute(f"UPDATE system.config SET value = '' where name = 'replication_strategy_fail_list'")
```

Comment:
I couldn't use `config_value_context` (I think), because I am updating `system.config` in the test's body, so I just clear system.config at the end.

Reply:
config_value_context is a context manager, i.e., you can do:

```python
     # this code uses the default configuration:
     ...
     with config_value_context(cql, 'replication_stategy_warn_list', 'network_topology'):
          # this code uses modified configuration:
          ...
     # when the context exits, we are using the default configuration again:
     ...
```

In any case, please don't "clear" the config at the end, because you don't know what it was before. The best thing is to return it to what it was. And this is what config_value_context does.

By the way, if you want your test to have many configuration changes in the middle, and at the end all of them are undone, you can do this with python's ExitStack to hold a stack of contextmanagers that need to be undone. For example:

```python
from contextlib import ExitStack
with ExitStack() as config_modifications:
      # this code uses the default config
      ....
       # this code uses modified config:
       config_modifications.enter_context(config_value_context(cql, 'replication_stategy_warn_list', 'network_topology')
       ...
       # make another config modification, the previous one is still in force:
       config_modifications.enter_context(config_value_context(cql, 'replication_stategy_error_list', 'network_topology')
       ...
# When the "with" above ends, all the contexted entered into config_modifications will be undone. The following code will again use the default configuration:
....
```

Reply:
Thanks, adjusted the code accordingly.